boards/arm/udoo_neo_full_m4/doc/udoo_neo_full.rst

```diff
@@ -0,0 +1,216 @@
+.. _udoo_neo_full_m4:
+
+UDOO Neo Full
+#############
+
+Overview
+********
+
+UDOO Neo Full is an open source Arduino Uno compatible single board computer.
+It is equipped with an NXP |reg| i.MX 6SoloX hybrid multicore processor
+composed by one ARM |reg| Cortex-A9 core running up to 1 GHz and one Cortex-M4
+core running up to 227 MHz for high CPU performance and real-time response.
+Zephyr was ported to run on the Cortex-M4 core. In a later release, it will
+also communicate with the Cortex-A9 core (running Linux) via RPMsg.
+
+.. image:: udoo_neo_full.jpg
```

Comment:
The ``image`` directive doesn't allow a caption (hence the error about content not being allowed.  Change this to use the ``figure`` directive (everything else is the same), which does allow content (for the caption).